apollo-compiler/src/test/graphql/com/example/named_fragment_inside_inline_fragment/GetHero.kt

```diff
@@ -0,0 +1,114 @@
+// AUTO-GENERATED FILE. DO NOT MODIFY.
+//
+// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
+// It should not be modified by hand.
+//
+package com.example.named_fragment_inside_inline_fragment
+
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.internal.QueryDocumentMinifier
+import com.apollographql.apollo.api.internal.ResponseAdapter
+import com.example.named_fragment_inside_inline_fragment.adapter.GetHero_ResponseAdapter
+import com.example.named_fragment_inside_inline_fragment.fragment.CharacterAppearsIn
+import com.example.named_fragment_inside_inline_fragment.fragment.CharacterName
+import com.example.named_fragment_inside_inline_fragment.type.Episode
+import kotlin.String
+import kotlin.Suppress
+import kotlin.collections.List
+
+@Suppress("NAME_SHADOWING", "UNUSED_ANONYMOUS_PARAMETER", "LocalVariableName",
+    "RemoveExplicitTypeArguments", "NestedLambdaShadowedImplicitParameter", "PropertyName",
+    "RemoveRedundantQualifierName")
+class GetHero : Query<GetHero.Data> {
+  override fun operationId(): String = OPERATION_ID
+
+  override fun queryDocument(): String = QUERY_DOCUMENT
+
+  override fun variables(): Operation.Variables = Operation.EMPTY_VARIABLES
+
+  override fun name(): String = OPERATION_NAME
+
+  override fun adapter(): ResponseAdapter<Data> = GetHero_ResponseAdapter
+  /**
+   * The query type, represents all of the entry points into our object graph
+   */
+  data class Data(
+    val hero: Hero?
+  ) : Operation.Data {
+    /**
+     * A character from the Star Wars universe
+     */
+    interface Hero {
+      val __typename: String
+
+      interface Character : Hero {
+        override val __typename: String
+
+        interface Character : Hero.Character, CharacterName, CharacterAppearsIn {
+          override val __typename: String
+
+          /**
+           * The name of the character
+           */
+          override val name: String
+
+          /**
+           * The movies this character appears in
+           */
+          override val appearsIn: List<Episode?>
+        }
+      }
+
+      data class CharacterHero(
+        override val __typename: String,
+        /**
+         * The name of the character
+         */
+        override val name: String,
+        /**
+         * The movies this character appears in
+         */
+        override val appearsIn: List<Episode?>
+      ) : Hero, Character, Character.Character, CharacterName, CharacterAppearsIn
+
+      data class OtherHero(
+        override val __typename: String
+      ) : Hero
```

Comment:
It bugs me that we are generating 2 interfaces and 2 data classes here when we know there is only one type condition. Could we generate just one data class ? That'll make the generated code easier to read and also faster to compile. And I think this doesn't lose any feature?